packages/webapp/src/components/Animals/DetailCards/General.tsx

```diff
@@ -40,10 +40,29 @@ import {
   Option as AnimalSelectOption,
   AnimalBreedSelect,
 } from '../AddAnimalsFormCard/AnimalSelect';
+import { generateUniqueAnimalId, parseUniqueDefaultId } from '../../../util/animal';
+
+// Add Animals Flow
+type SingleAnimalTypeUses = Option[DetailsFields.USE][];
+
+// Single Animal View
+type AnimalTypeToUsesMapping = {
+  default_type_id: number;
+  uses: Option[DetailsFields.USE][];
+}[];
+
+type UseOptions = SingleAnimalTypeUses | AnimalTypeToUsesMapping;
+
+// type guard
+const isAnimalTypeUsesDictionary = (
+  useOptions: UseOptions,
+): useOptions is AnimalTypeToUsesMapping => {
+  return 'default_type_id' in useOptions[0];
```

Comment:
This line is causing an error for me sometimes, saying `Uncaught TypeError: Cannot use 'in' operator to search for 'default_type_id' in undefined`

Reply:
I started seeing that a lot! The whole type guard is removed now so shouldn't be possible.